### DISCUSSION TITLE
Improve shape of `ReplyCurveLayoutElement`

### DISCUSSION THREAD

**Discussion by Nerixyz** [Chatterino/chatterino2#4009] [Ideas]
The reply curve looks odd. I think that's because the curve doesn't lay on a circle:

<img src="https://user-images.githubusercontent.com/19953266/192119100-1919ac23-a216-4422-9876-6f213d7f6d87.svg" alt="old" width="200"/>

One change is to have a quadratic curve and a line:

<img src="https://user-images.githubusercontent.com/19953266/192119176-887ecc1d-843a-4f98-895d-cf0aaf43400d.svg" alt="new" width="200"/>

<details>
<summary>Diff</summary>

```diff
diff --git a/src/messages/layouts/MessageLayoutElement.cpp b/src/messages/layouts/MessageLayoutElement.cpp
index 492a4c8b..de03f1c3 100644
--- a/src/messages/layouts/MessageLayoutElement.cpp
+++ b/src/messages/layouts/MessageLayoutElement.cpp
@@ -461,11 +461,13 @@ void ReplyCurveLayoutElement::paint(QPainter &painter)
     qreal bottom = paintRect.bottom() - this->neededMargin_;
     QPointF startPoint(left, bottom);
     QPointF controlPoint(left, top);
+    QPointF circleEndPoint(left + (bottom - top), top);
     QPointF endPoint(paintRect.right(), top);
 
     // Create curve path
     bezierPath.moveTo(startPoint);
-    bezierPath.quadTo(controlPoint, endPoint);
+    bezierPath.quadTo(controlPoint, circleEndPoint);
+    bezierPath.lineTo(endPoint);
 
     // Render curve
     painter.setPen(this->pen_);
```
</details>

This doesn't exactly lay on the circle, but it's really tiny so no one should notice.

You can however use `arcTo` to draw the arc of the circle:

<img src="https://user-images.githubusercontent.com/19953266/192119218-a13644a9-8522-4969-989a-7c1d6df812ab.svg" alt="new-arc" width="200"/>

<details>
<summary>Diff</summary>

```diff
diff --git a/src/messages/layouts/MessageLayoutElement.cpp b/src/messages/layouts/MessageLayoutElement.cpp
index 492a4c8b..f4ba5d01 100644
--- a/src/messages/layouts/MessageLayoutElement.cpp
+++ b/src/messages/layouts/MessageLayoutElement.cpp
@@ -460,12 +460,13 @@ void ReplyCurveLayoutElement::paint(QPainter &painter)
     qreal left = paintRect.left() + this->neededMargin_;
     qreal bottom = paintRect.bottom() - this->neededMargin_;
     QPointF startPoint(left, bottom);
-    QPointF controlPoint(left, top);
     QPointF endPoint(paintRect.right(), top);
 
     // Create curve path
     bezierPath.moveTo(startPoint);
-    bezierPath.quadTo(controlPoint, endPoint);
+    bezierPath.arcTo(left, top, (bottom - top) * 2, (bottom - top) * 2, 180,
+                     -90);
+    bezierPath.lineTo(endPoint);
 
     // Render curve
     painter.setPen(this->pen_);
```
</details>

In my opinion the green curve would be enough since you can't see the difference between the green and blue curves when they're small.

**Comment by leon-richardt** [comment]
If we make a principled argument ("the curve should lay on a circle") to justify a change, it would make more sense to me to use the curve that _actually_ lays on a circle, i.e., the blue one.

**Comment by dnsge** [comment]
Related PR: #4077
